clusterloader2/state/namespaces.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// InstancesState represents state of object replicas.
+type InstancesState struct {
+	Number int32
+	Object api.Object
+}
+
+type instancesIdentifier struct {
+	basename   string
+	objectKind string
```

Comment:
Instead of adding these separate string fields for "objectKind", "apiGroup" - could you reuse the "GroupKind" built-in type in k8s. Ref - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/group_version.go#L57_L64

Reply:
TBH, I don't see what are the advantages of this change. instancesIdentifier is only a local structure and there is no method of GroupKind needed here.